ExadelTimeTrackingSystem/ExadelTimeTrackingSystem.BusinessLogic/Services/TaskService.cs

```diff
@@ -39,10 +42,43 @@ public async Task<TaskDTO> GetByIdAsync(Guid id)
             return _mapper.Map<TaskDTO>(task);
         }
 
-        public async Task<List<TaskDTO>> GetTasksOnDateAsync(DateTime date)
+        public async Task<List<TaskDTO>> GetOnDateAsync(DateTime date)
         {
-            var tasks = await _repository.GetTasksOnDateAsync(date);
+            var tasks = await _repository.GetOnDateAsync(date);
             return _mapper.Map<List<TaskDTO>>(tasks);
         }
+
+        public Task DeleteAsync(Guid id)
+        {
+            return _repository.DeleteAsync(id);
+        }
+
+        public async Task<TaskDTO> UpdateAsync(TaskDTO taskDto)
+        {
+            var task = _mapper.Map<Data.Models.Task>(taskDto);
+            await _repository.UpdateAsync(task);
+            return taskDto;
+        }
+
+        public Task ApproveAsync(DateTime date, Guid projectId, Guid employeeId)
+        {
+            return _repository.ApproveAsync(date, projectId, employeeId);
+        }
+
+        public async Task<List<CreateTaskDTO>> BulkCreateAsync(BulkCreateTaskDTO bulkCreateTaskDto)
+        {
+            var list = new List<CreateTaskDTO>();
+            var projectName = await _projectService.GetNameAsync(bulkCreateTaskDto.Task.ProjectId);
+            var newTasks = bulkCreateTaskDto.Dates.Select(date =>
+            {
+                var task = _mapper.Map<Data.Models.Task>(bulkCreateTaskDto.Task);
+                task.Date = date;
+                task.ProjectName = projectName;
+                return task;
+            }).ToList();
+
+            await _repository.BulkCreateAsync(newTasks);
+            return list;
```

Comment:
you're returning now just empty list. please try to check the method from swagger and check the response